demos/mask_rcnn_demo/cpp/main.cpp

```diff
@@ -45,240 +49,327 @@ bool ParseAndCheckCommandLine(int argc, char *argv[]) {
-        // -----------------------------------------------------------------------------------------------------
+    const float PROBABILITY_THRESHOLD = 0.2f;
+    // threshold used to determine whether mask pixel corresponds to object or to background
+    const float MASK_THRESHOLD = 0.5f;
 
-        // ---------------------------Postprocess output blobs--------------------------------------------------
-        const auto do_blob = infer_request.GetBlob(FLAGS_detection_output_name.c_str());
-        InferenceEngine::LockedMemory<const void> doBlobMapped =
-            InferenceEngine::as<InferenceEngine::MemoryBlob>(do_blob)->rmap();
-        const auto do_data  = doBlobMapped.as<float*>();
+    size_t box_stride = W * H * C;
 
-        const auto masks_blob = infer_request.GetBlob(FLAGS_masks_name.c_str());
-        InferenceEngine::LockedMemory<const void> masksBlobMapped =
-            InferenceEngine::as<InferenceEngine::MemoryBlob>(masks_blob)->rmap();
-        const auto masks_data = masksBlobMapped.as<float*>();
+    std::map<size_t, size_t> class_color;
 
-        const float PROBABILITY_THRESHOLD = 0.2f;
-        const float MASK_THRESHOLD = 0.5f;  // threshold used to determine whether mask pixel corresponds to object or to background
-        // amount of elements in each detected box description (batch, label, prob, x1, y1, x2, y2)
-        IE_ASSERT(do_blob->getTensorDesc().getDims().size() == 2);
-        size_t BOX_DESCRIPTION_SIZE = do_blob->getTensorDesc().getDims().back();
+    std::vector<cv::Mat> output_images;
+    for (const auto& img : images) {
+        output_images.push_back(img.clone());
+    }
 
-        const InferenceEngine::TensorDesc& masksDesc = masks_blob->getTensorDesc();
-        IE_ASSERT(masksDesc.getDims().size() == 4);
-        size_t BOXES = getTensorBatch(masksDesc);
-        size_t C = getTensorChannels(masksDesc);
-        size_t H = getTensorHeight(masksDesc);
-        size_t W = getTensorWidth(masksDesc);
+    // Iterating over all boxes
+    for (size_t box = 0; box < BOXES; ++box) {
+        float* box_info = boxes_data + box * BOX_DESCRIPTION_SIZE;
+        auto batch = static_cast<int>(box_info[0]);
 
+        if (batch < 0)
+            break;
+        if (batch >= static_cast<int>(modelBatchSize))
+            throw std::logic_error("Invalid batch ID within detection output box");
 
-        size_t box_stride = W * H * C;
+        float prob = box_info[2];
 
-        std::map<size_t, size_t> class_color;
+        float x1 = std::min(std::max(0.0f, box_info[3] * images[batch].cols), static_cast<float>(images[batch].cols));
+        float y1 = std::min(std::max(0.0f, box_info[4] * images[batch].rows), static_cast<float>(images[batch].rows));
+        float x2 = std::min(std::max(0.0f, box_info[5] * images[batch].cols), static_cast<float>(images[batch].cols));
+        float y2 = std::min(std::max(0.0f, box_info[6] * images[batch].rows), static_cast<float>(images[batch].rows));
```

Comment:
Please reuse https://github.com/openvinotoolkit/open_model_zoo/blob/24346485099c98f368afb2cb001eaf34925e0018/demos/common/cpp/utils/include/utils/common.hpp#L58

Reply:
ok, changed to clamp